### PR TITLE
Added handling SIGTERM signal for proper stopping dedicated server

### PR DESCRIPTION
**Pull request by SNMetamorph**

### CODE REVIEW

File: engine/platform/posix/sys_posix.c [hunk 1]
```diff
@@ -145,6 +146,16 @@ void Posix_Daemonize( void )
...
 }
 
+static void Posix_SigtermCallback( int signum )
+{
+	Sys_Quit();
+}
+
+void Posix_SetupSigtermHandling( void )
+{
+	signal( SIGTERM, Posix_SigtermCallback );
```
Comment (a1batross) [inline engine/platform/posix/sys_posix.c:156]: Better use `sigaction(2)`, it's more portable than `signal(2)`.

**Review by a1batross — CHANGES_REQUESTED** [review]

File: engine/platform/posix/sys_posix.c (hunk 1 above)
Comment (a1batross) [reply engine/platform/posix/sys_posix.c:156]: The example is in `crashhandler.c` btw.

Comment (SNMetamorph) [reply engine/platform/posix/sys_posix.c]: Ok, I did it.

**Comment by SNMetamorph** [comment]
Unfortunately, PS Vita knows nothing about `sigaction`, so I decided to add `#if` condition to skip it there. Anyway, there is no possibility on PS Vita to properly working with processes and signals like on other normal OSes.

**Review by a1batross — APPROVED** [review]

**Pull request merged**